lib/transport.ex

```diff
@@ -3,8 +3,49 @@ defmodule Janus.Transport do
   @type state :: any
   @type payload :: map
 
+  @doc """
+  Creates transport specific connection and preserves it in state.
+
+  Returns either `{:ok, state}` indicating valid transport state
+  or `{:error, reason}` indicating connection failure.
+  """
   @callback connect(any) :: {:ok, state} | {:error, reason}
+
+  @doc """
+  Synchronously sends given payload via previously created transport connection respecting given timeout.
+
+  ## Arguments
+  * `payload` - arbitrary map structure to be sent via transport
+  * `timeout` - time after which `send/3` call should return imidiately with an error
+  * `state` - state structure containing valid transport data necessary to send payload
+
+  ## Returns
+  `{:ok, state}` on successful payload delivery
+  `{:error, reason, state}` on delivery failure
+  """
   @callback send(payload, timeout, state) :: {:ok, state} | {:error, reason, state}
-  @callback handle_info(any, state) ::
+
+  @doc """
+  Handles arbitrary data to be processed by transport implementation.
```

Comment:
```suggestion
  Handles messages received by connection process
```

This callback might need further changes, but outside this PR